src/bluez.c

```diff
@@ -712,6 +713,8 @@ static int bluez_register_a2dp_endpoint(
 
 final:
 	g_free(path);
+	g_variant_builder_unref(properties);
```

Comment:
If code from the line [679](https://github.com/Arkq/bluez-alsa/pull/119/files#diff-8c15b7742dad8d123d1be2d1c3954505R679) will be executed, then we've got g_variant_builder_unref() called with an uninitialized variable. And this comment applies to all your changes.

Many thanks for pointing out that the memory allocated with the g_variant_builder_new() needs to be freed. However, I think, that in this case it will be better to call g_variant_builder_init() instead - it is simpler and avoids heap memory allocation.

Reply:
@Arkq Yes, stack variable should be fine. Shall I change it to g_variant_builder_init ?

Reply:
Yes, if you might :)